huntsman_dust/power_spectrum.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+import util_rad_profile
+from scipy import fftpack
+from scipy import optimize
+import matplotlib.pyplot as plt
+import huntsman_dust.util_plot as util_plot
+from huntsman_dust.detect_mask_sources import detect_mask_sources
+
+
+def power_spectrum(image):
+    image = image.astype(float)
+    psd2D = np.abs(fftpack.fftshift(fftpack.fft2(image)))**2
+    psd1D = util_rad_profile.azimuthalAverage(psd2D)
+    return psd1D
+
+
+if __name__ == '__main__':
+    image, img_masked, _, _ = detect_mask_sources(image_path='/Users'
+                                                  + '/amanchokshi/Desktop'
+                                                  + '/Huntsman/Data'
+                                                  + '/ngc6822_r.fits',
+                                                  slice_in=1500,
+                                                  slice_out=3000,
+                                                  obj_name='NGC6822',
+                                                  Ra=296.234,
+                                                  Dec=-14.797,
+                                                  obj_radius=10)
+
+    psd1D_im = power_spectrum(image)
```

Comment:
It looks like this test is run on real images. Perhaps provide an example where you run it on fake images, which is kind of the next step anyways.

Reply:
to find PS of fake image python power_spectrum.py
Otherwise, a image path must be supplied using the --img_path flag